compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/ExpressionCodegen.kt

```diff
@@ -599,13 +640,21 @@ class ExpressionCodegen(
                 mv,
                 this
             )
-            mv.newarray(boxType(elementIrType.asmType))
+            mv.newarray(typeMapper.boxType(elementIrType))
         } else {
-            val type = typeMapper.mapType(arrayType)
-            mv.newarray(correctElementType(type))
+            mv.newarray(correctElementType(arrayType.asmType))
         }
     }
 
+    // Copied from FinallyBlocksLowering
+    private val IrReturnTarget.returnType: IrType
```

Comment:
The problem with this is that `returnType` is specific to the JVM backend in the case of constructors.

Changing the IR by adding new fields or methods also needs input from JS_IR and native, so whenever possible I'd rather avoid it and just add extension methods. Otherwise I agree that this should be a property.

Reply:
Done